llvm/include/llvm/CodeGen/GlobalISel/IRTranslator.h

```diff
@@ -235,26 +235,26 @@ class IRTranslator : public MachineFunctionPass {
   bool translateStore(const User &U, MachineIRBuilder &MIRBuilder);
 
   /// Translate an LLVM string intrinsic (memcpy, memset, ...).
-  bool translateMemFunc(const CallInst &CI, MachineIRBuilder &MIRBuilder,
+  bool translateMemFunc(const CallBase &CI, MachineIRBuilder &MIRBuilder,
```

Comment:
I doubt we need most of these CallInst -> CallBase changes - you're not letting you use this with arbitrary intrinsics

Reply:
I'd say it makes it harder to reason about. If the function is only used by CallInst and will only ever handle CallInst, using CallBase is misleading. I'd expect this case to use some common helper functions without being forced through the existing ordinary call visitors

Reply:
I reimplemented callbr in SelectionDAGBuilder and IRTranslator with this in mind, thanks for the feedback (as always ^^)